monkey/tests/unit_tests/monkey_island/cc/resources/test_agent_plugins.py

```diff
@@ -69,17 +74,33 @@ def test_get_plugins__not_found_if_name_does_not_exist(flask_client):
     ["DummyType", "ExploiteR"],
 )
 def test_get_plugins__not_found_if_type_is_invalid(flask_client, type_):
-    resp = flask_client.get(get_url_for_resource(AgentPlugins, plugin_type=type_, name="name"))
+    resp = flask_client.get(
+        get_url_for_resource(AgentPlugins, host_os="linux", plugin_type=type_, name=FAKE_NAME)
+    )
 
     assert resp.status_code == HTTPStatus.NOT_FOUND
 
 
 def test_get_plugins__server_error(flask_client, agent_plugin_repository):
-    def raise_retrieval_error(plugin_type, name):
+    def raise_retrieval_error(host_os, plugin_type, name):
         raise RetrievalError
 
     agent_plugin_repository.get_plugin = raise_retrieval_error
 
-    resp = flask_client.get(get_url_for_resource(AgentPlugins, plugin_type="Payload", name="name"))
+    resp = flask_client.get(
+        get_url_for_resource(AgentPlugins, host_os="linux", plugin_type=FAKE_TYPE, name=FAKE_NAME)
+    )
 
     assert resp.status_code == HTTPStatus.INTERNAL_SERVER_ERROR
+
+
+@pytest.mark.parametrize(
+    "os_",
+    ["macos", "darwin"],
+)
+def test_get_plugins__not_found_if_os_is_invalid(flask_client, os_):
+    resp = flask_client.get(
+        get_url_for_resource(AgentPlugins, host_os=os_, plugin_type=FAKE_TYPE, name=FAKE_NAME)
+    )
+
+    assert resp.status_code == HTTPStatus.NOT_FOUND
```

Comment:
This test should result in some kind of 4xx error, since "macos" and "darwin" are invalid OSes.

Reply:
NVM. I misread this.